src/main/java/Duke.java

```diff
@@ -1,10 +1,112 @@
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+    private Ui ui;
+    private Storage storage;
+    private TaskList tasks;
+    private Parser parser;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        parser = new Parser();
+
+        try {
+            tasks = new TaskList(storage.loadFromFile());
+        } catch (IOException e) {
+             System.out.println(e);
+        }
+
+    }
+
+    public void run() throws IOException {
+        ui.printWelcomeMessage();
+        storage.loadFromFile();
+        Scanner in = new Scanner(System.in);
+        String input = in.nextLine(); // Get string input
+
+        while (!input.equals("bye")) {
+            String[] words = input.split(" ", 2); // Split command from rest of sentence
+            try {
+                storage.writeToFile(tasks.getTaskList());
+                manageCommand(input, words);
+            } catch (InvalidTaskIndexException e) {
+                System.out.println("\t ☹ OOPS!!! The task index you've entered is invalid.");
+            } catch (EmptyTaskException e) {
+                System.out.println("\t ☹ OOPS!!! The description of the task cannot be empty.");
+            } catch (InvalidCommandException e) {
+                System.out.println("\t ☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+                System.out.println("\t Type 'help' for the list of commands available.");
+            } catch (InvalidDeadlineException e) {
+                System.out.println("\t ☹ OOPS!!! The description of a deadline must be in this format: ");
+                System.out.println("\t deadline <task name> /by <date/time>");
+            } catch (InvalidEventException e) {
+                System.out.println("\t ☹ OOPS!!! The description of an event must be in this format: ");
+                System.out.println("\t event <task name> /at <date/time>");
+            } catch (IOException e) {
+                System.out.println("\t ☹ OOPS!!! File is not found!");
+            } catch (EmptyListException e) {
+                System.out.println("\t ☹ OOPS!!! Your task list is empty! Try adding a task first!");
+            } catch (NumberFormatException e) {
+                System.out.println("\t ☹ OOPS!!! You entered an invalid number!");
+            }
```

Comment:
Good attempt in handling exceptions! However you may want to categorise the exceptions as one IllegalDukeException and assign a specific message to each case. This can help to reduce the line of codes and add more SLAP to your code.